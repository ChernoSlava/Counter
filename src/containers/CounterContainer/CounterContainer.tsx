import React, { useContext } from "react";
import { Counter, CounterType } from "@components"
import { CounterContext } from "@contexts";

export const CounterContainer: React.FC = (): JSX.Element => {
    const {state} = useContext(CounterContext);
    let type: CounterType = 'lesszero';

    if ( state.counterValue === 0 ) {
        type = 'zero';
    } else if ( state.counterValue > 0 ) {
        type = 'abovezero';
    }

    return (
        <Counter title={String(state.counterValue)} type={type}/>
    )
}