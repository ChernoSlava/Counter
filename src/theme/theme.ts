export const theme = {
  colors: {
    grey10: 'darkgrey',
    aqua10: 'aquamarine',
    red10: 'indianred',
    red20: 'red',
    black: 'black',
    green10: 'green',
  },
};

export type ThemeType = typeof theme;
