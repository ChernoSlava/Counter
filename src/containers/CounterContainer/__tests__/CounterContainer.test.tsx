import { CounterContainer } from "../CounterContainer";
import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Store, AnyAction } from '@reduxjs/toolkit';

import { theme } from '../../../theme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { InitialCounter } from "@store";

const mockStore = configureStore();

const getUI = (store: Store<any, AnyAction>): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <CounterContainer />
      </Provider>
    </ThemeProvider>
  );
};

describe('CounterContainer test', () => {
  it('should render zero', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
    }),
    ))
    expect(container).toMatchSnapshot();
  });

  it('should render lesszero', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
      value: -1,
    }),
    ))
    expect(container).toMatchSnapshot();
  });

  it('should render abovezero', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
      value: 4,
    }),
    ))
    expect(container).toMatchSnapshot();
  });
})
