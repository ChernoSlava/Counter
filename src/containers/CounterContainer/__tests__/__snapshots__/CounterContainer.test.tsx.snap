// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterContainer test should render abovezero 1`] = `
.c0 {
  font-size: 24px;
  color: black;
  font-weight: bold;
  color: green;
}

<div>
  <div
    class="c0"
    type="abovezero"
  >
    4
  </div>
</div>
`;

exports[`CounterContainer test should render lesszero 1`] = `
.c0 {
  font-size: 24px;
  color: black;
  font-weight: bold;
  color: indianred;
}

<div>
  <div
    class="c0"
    type="lesszero"
  >
    -1
  </div>
</div>
`;

exports[`CounterContainer test should render zero 1`] = `
.c0 {
  font-size: 24px;
  color: black;
  font-weight: bold;
  color: black;
}

<div>
  <div
    class="c0"
    type="zero"
  >
    0
  </div>
</div>
`;
